.github/workflows/build.yaml

```diff
@@ -34,8 +38,21 @@ jobs:
       - name: Run build
         run: npm run package
 
+      # Generate .vsix package for the extension
+      - name: Generate .vsix package
+        run: |
+          npm install -g @vscode/vsce
+          vsce package
+
+      - name: Upload VSIX artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vscode-granite-vsix-${{ runner.os }}-${{ github.run_id }}  # Unique name with OS and run ID
+          path: "./vscode-granite-*.vsix"
+
+
       # Run tests
       - name: Run Test
         uses: coactions/setup-xvfb@b6b4fcfb9f5a895edadc3bc76318fae0ac17c8b3 #v1.0.1
         with:
-          run: npm test
\ No newline at end of file
+          run: npm test
```

Comment:
Create a rolling dev release for the main branch:

      - name: Upload latest vsix
        if: runner.os == 'Linux' && github.ref == 'refs/heads/main'
        uses: marvinpinto/action-automatic-releases@latest
        with:
          repo_token: "${{secrets.GITHUB_TOKEN}}"
          automatic_release_tag: "latest"
          prerelease: true
          title: "Development Build"
          files: |
            *.vsix